src/main/java/scientific/Statistics.java

```diff
@@ -8,132 +8,102 @@
 
 package scientific;
 
-import java.util.ArrayList;
-import java.util.Arrays;
-import java.util.HashMap;
-import java.util.Map;
+import java.util.*;
 
 public class Statistics {
 
-	public static double mean(double[] data){
-		double total = 0;
-		for(double d : data){
-			total = total + d;
+	private static void checkInputHasElements(double[] inputArray) {
+		if(inputArray.length == 0) {
+			throw new IllegalArgumentException("The array of values must contain at least one element");
 		}
+	}
 
-		return total/data.length;
+	public static double mean(double[] valuesToCalculateMeanOf){
+		checkInputHasElements(valuesToCalculateMeanOf);
+		double total = Arrays.stream(valuesToCalculateMeanOf).sum();
```

Comment:
There's also an option to calculate the average using streams too. The only issue with this is that it returns an OptionalDouble which you then have to check for null or not. It makes the function a bit longer but may be more reliable.
```
OptionalDouble average = Arrays.stream(valuesToCalculateMeanOf).average();
if(average.isPresent) {
    return average.getAsDouble;
}
throw new IllegalStateException("Unable to calculate average, OptionalDouble null")
````

Reply:
This is fine.